src/web-ui/src/components/authentication/new-password/new-password.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AuthenticationApiService } from '../../../services/api/authentication-api.service';
+import { AbstractControl, FormBuilder, FormControl, Validators } from '@angular/forms';
+import { CommonService } from '../../../services/common.service';
+import { ToastType } from '../../../services/toast.service';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { UserProfile } from '../../../models/user-profile.model';
+
+
+@Component({
+  selector: 'new-password',
+  templateUrl: './new-password.component.html',
+  styleUrls: ['./new-password.component.scss']
+})
+export class NewPasswordComponent implements OnInit, OnDestroy {
+  form = new FormBuilder().group({
+    password: new FormControl(null, [Validators.required, Validators.minLength(5)]),
+    password2: new FormControl(null, [Validators.required, Validators.minLength(5)])
+  });
+  subscription: Subscription = new Subscription();
+  token = '';
+  userProfile: UserProfile | null = null;
+
+  constructor(private commonService: CommonService,
+              private router: Router,
+              private apiService: AuthenticationApiService) {}
+
+  ngOnInit() {
+    this.getUserByToken();
+    this.password2ValueSubscription();
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  getUserByToken() {
+    const url = this.router.routerState.snapshot.url;
+    this.token = url.substring(url.indexOf('=') + 1, url.length);
+    this.apiService.getUserByResetToken(this.token).subscribe({
+      next: (userProfile) => {
+        this.userProfile = userProfile;
+      },
+      error: () => {
+        this.commonService.goRoute('');
+        this.commonService.showToast('components.new-password.toasts.token-expired', ToastType.ERROR);
+      }
+    });
+  }
+
+  password2ValueSubscription() {
+    this.subscription = this.getPassword2().valueChanges.subscribe({
+      next: (password2Value) => {
+        const passwordValue = this.getPassword().value;
+        const password2Control = this.getPassword2();
+        if (passwordValue && password2Value && passwordValue !== password2Value) {
+          password2Control?.setErrors({ doesntMatch: true });
+        } else {
+          password2Control?.setErrors(null);
+        }
+      }
+    });
+  }
+
+  setNewPassword() {
+    this.form.markAllAsTouched();
+    if (this.form.valid && this.userProfile) {
+      this.apiService.setNewPassword(this.userProfile?.id, this.getPassword().value).subscribe({
```

Comment:
Na tym etapie nie mamy użytkownika. Nie będziesz miał skąd otrzymać jego id. 
Moim zdaniem nie powinniśmy mieć czegoś takiego jak pobieranie profilu po tokenie.
Dlatego do backendu powinieneś przesłać token oraz hasło.

Reply:
Tak też pierwotnie wykonałem, ale nie miałem bladego pojęcia jak to zrobisz i założyłem z userprofilem, poprawię ❣️